src/flashFormatter/FlashFormatterInterface.h

```diff
@@ -0,0 +1,30 @@
+/*
+  Copyright (c) 2025 Arduino SA
+
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+#pragma once
+#include <Arduino.h>
+
+class FlashFormatterClass {
+public:
+  virtual ~FlashFormatterClass() = default;
+  virtual bool checkandFormatPartition() {
+    if(checkPartition()){
+      return true;
+    }
+
+    if(!formatPartition()){
+      return false;
+    }
+
+    return checkPartition();
+  }
+
+protected:
+  virtual bool checkPartition() { return true; };
+  virtual bool formatPartition() { return true; };
```

Comment:
Why not pure virtual functions? Shouldn't this be an interface?

Reply:
Yeah maybe the name of the file is misleading, but in my idea this class is the base/default class from which derive. So in the sketch I don't have to use ifdef according to platform

Reply:
should we call it FlashFormatterBase.h ? @fabik111 @andreagilardoni

Reply:
It sounds better to me